### ISSUE TITLE
now doesn't account for DST

### ISSUE THREAD

**Issue by Bobselp** [jinzhu/now#13]
Hey,

I'm in central Europe, and we currently still have daylight saving time (timezone CEST instead of CET).
now doesn't seem to account for the extra hour, so `now.BeginningOfYear()` currently returns `2014-12-31 23:00:00 +0100 CET` instead of midnight.

Greets,
Bobselp

**Comment by wrouesnel** [comment]
+1

Just ran into this myself - had a program which uses beginning and end of month stop working because multi-month spans were being returned due to BeginningOfMonth.

**Comment by SDkie** [comment]
@Bobselp @wrouesnel: http://www.timeanddate.com/time/zones/cest
it says "CEST is currently not used at all"?

**Comment by Bobselp** [comment]
@SDkie CEST was over on October 25th. It will begin again on March 27th 2016.

**Comment by PolyviosT** [comment]
As @Bobselp said this issue just appeared again. If you try to do today : `now.BeginningOfDay()` you will get 2016-03-26 23:00:00 +0000 WET

**Comment by kmanley** [comment]
+1 this bit me today too. 

```
package main
import (
	"fmt"
	"time"
	"github.com/jinzhu/now"
)

func main() {
	fmt.Println(now.New(time.Date(2016, 11, 6, 0, 0, 0, 0, time.Local)).EndOfMonth())
	fmt.Println(now.New(time.Date(2016, 11, 7, 0, 0, 0, 0, time.Local)).EndOfMonth())
}
```

results in

```
2016-11-30 23:59:59.999999999 -0500 EST
2016-12-01 00:59:59.999999999 -0500 EST
```

**Comment by dolmen** [comment]
Mandatory link because the implementation of this library wrongly assumes that days are always 24h long:
http://infiniteundo.com/post/25509354022/more-falsehoods-programmers-believe-about-time

**Comment by jinzhu** [comment]
Fixed by PR #21 , sorry for the problem.
